userspace/libscap/engine/bpf/scap_bpf.c

```diff
@@ -1667,13 +1683,64 @@ static int32_t unsupported_config(struct scap_engine_handle engine, const char*
 	return SCAP_FAILURE;
 }
 
+static int32_t scap_bpf_handle_tp_mask(struct scap_engine_handle engine, uint32_t op, uint32_t tp)
+{
+	struct bpf_engine *handle = engine.m_handle;
+
+	int prg_idx = -1;
+	for (int i = 0; i < handle->m_bpf_prog_cnt; i++)
+	{
+		const tp_values val = tp_from_name(handle->m_bpf_progs[i].name);
+		if (val == tp)
+		{
+			prg_idx = i;
+			break;
+		}
+	}
+
+	// We want to unload a never loaded tracepoint
+	if (prg_idx == -1 && op != SCAP_TPMASK_SET)
+	{
+		return SCAP_SUCCESS;
+	}
+	// We want to load an already loaded tracepoint
+	if (prg_idx >= 0 && op != SCAP_TPMASK_UNSET)
+	{
+		return SCAP_SUCCESS;
+	}
+
+	if (op == SCAP_TPMASK_UNSET)
+	{
+		// Algo:
+		// Close the event and tracepoint fds,
+		// reduce number of prog cnt
+		// move left remaining array elements
+		// reset last array element
+		close_prog(&handle->m_bpf_progs[prg_idx]);
+		handle->m_bpf_prog_cnt--;
+		size_t byte_size = (handle->m_bpf_prog_cnt - prg_idx) * sizeof(handle->m_bpf_progs[prg_idx]);
+		if (byte_size > 0)
+		{
+			memmove(&handle->m_bpf_progs[prg_idx], &handle->m_bpf_progs[prg_idx + 1], byte_size);
+		}
+		memset(&handle->m_bpf_progs[handle->m_bpf_prog_cnt], 0, sizeof(handle->m_bpf_progs[handle->m_bpf_prog_cnt]));
+		return SCAP_SUCCESS;
+	}
+
+	uint64_t api_version_p;
+	uint64_t schema_version_p;
+	scap_open_args oargs = {0};
+	oargs.tp_of_interest.tp[tp] = 1;
+	return load_bpf_file(handle, &api_version_p, &schema_version_p, &oargs);
```

Comment:
Reload bpf file with the new tp of interest.
It will just attach the new tracepoint, given the `already_attached` check that was added (mentioned before in this self-review).